src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -0,0 +1,300 @@
+            /// </summary>
+            /// <param name="value">The character array from which to write characters.</param>
+            /// <param name="index">The starting index in the character array from which to begin writing characters.</param>
+            /// <param name="count">The number of characters to write from the character array.</param>
+            public override void Write(char[] value, int index, int count)
+            {
+                ReadOnlySpan<char> charValuesToWrite = value.AsSpan().Slice(index, count);
+                this.StartWritingCharValuesInChunks(charValuesToWrite);
+            }
+
+            /// <summary>
+            /// Asynchronously writes a specified number of characters from the given character array to theODataUtf8JsonWriter.
+            /// </summary>
+            /// <param name="value">The character array from which to write characters.</param>
+            /// <param name="index">The starting index in the character array from which to begin writing characters.</param>
+            /// <param name="count">The number of characters to write from the character array.</param>
+            /// <returns>A task representing the asynchronous write operation.</returns>
+            public override async Task WriteAsync(char[] value, int index, int count)
+            {
+                ReadOnlyMemory<char> charValuesToWrite = buffer.AsMemory().Slice(index, count);
+                await this.StartWritingCharValuesInChunksAsync(charValuesToWrite);
+            }
+
+            /// <summary>
+            /// Writes the characters represented by the provided ReadOnlySpan in chunks, processing them for escaping if necessary, and flushing the writer if the buffer threshold is reached.
+            /// </summary>
+            /// <param name="value">The ReadOnlySpan containing the characters to be written.</param>
+            private void StartWritingCharValuesInChunks(ReadOnlySpan<char> value)
+            {
+                // Process the input value in chunks
+                for (int i = 0; i < value.Length; i += chunkSize)
+                {
+                    int remainingChars = Math.Min(chunkSize, value.Length - i);
+                    bool isFinalBlock = false;
+                    ReadOnlySpan<char> chunk = value.Slice(i, remainingChars);
+
+                    // Get unprocessed chars from the buffer.
+                    ReadOnlySpan<char> charsNotProcessedFromPreviousChunk = buffer;
+
+                    int firstIndexToEscape = this.jsonWriter.NeedsEscaping(chunk);
+
+                    // If the buffer is not empty, then we copy the chars from the buffer
+                    // to the current chunk being processed.
+                    if (!charsNotProcessedFromPreviousChunk.IsEmpty)
+                    {
+                        char[] combinedArray = new char[charsNotProcessedFromPreviousChunk.Length + chunk.Length];

```

Comment:
Would it be better to rent the byte array from `ArrayPool<byte>`?

Reply:
done.